### PR TITLE
Remove ShimmeredDetailsList from the DetailsList export

### PR DESCRIPTION
**Pull request by ThomasMichon**
PR #5374 incorrectly added `ShimmeredDetailsList` to the `DetailsList` export root, defeating the purpose of factoring it into its own component to reduce bundle cost.

### CODE REVIEW

**Comment by ThomasMichon** [comment]
@dzearing technically this is a breaking change, but it's the only way to fix the bundle issue without burdening downstream projects.

File: packages/office-ui-fabric-react/src/components/DetailsList/index.ts [hunk 1]
```diff
@@ -4,4 +4,3 @@ export * from './DetailsList';
 export * from './DetailsList.types';
 export * from './DetailsRow';
 export * from './DetailsRowCheck';
-export * from './ShimmeredDetailsList';
```
Comment (Vitalius1) [inline packages/office-ui-fabric-react/src/components/DetailsList/index.ts]: @ThomasMichon As a solution you should maybe add a `ShimmeredDetailsList.ts` file at the `packages/office-ui-fabric-react/src` level to serve as an entry point to this component which will contain `export ShimmeredDetailsList from './components/DetailsList/ShimmeredDetailslIst'`. Plus the build is breaking cause I am using this ShimmeredDetailsList in a Shimmer Example page. My solution should fix this build error and give us the result we want regarding the bundle size. I am not an expert when it comes to bundling so I leave this to you...

File: packages/office-ui-fabric-react/src/components/Shimmer/examples/Shimmer.Application.Example.tsx [hunk 2]
```diff
@@ -4,7 +4,7 @@ import * as React from 'react';
 import { BaseComponent } from 'office-ui-fabric-react/lib/Utilities';
 import { createListItems } from '../../../utilities/exampleData';
 import { IColumn, buildColumns, SelectionMode, Toggle } from 'office-ui-fabric-react/lib/index';
-import { ShimmeredDetailsList } from '../../DetailsList';
+import { ShimmeredDetailsList } from '../../DetailsList/ShimmeredDetailsList';
```
Comment (Vitalius1) [inline packages/office-ui-fabric-react/src/components/Shimmer/examples/Shimmer.Application.Example.tsx]: @ThomasMichon This is in code examples and any user reading the documentation will see it... I don't believe this shows them how to appropriately import this component. Shouldn't it be `import { ShimmeredDetailsList } from 'office-ui-fabric-react/lib/ShimmeredDetailsList';`? Read my previous comment.

Comment (Vitalius1) [reply packages/office-ui-fabric-react/src/components/Shimmer/examples/Shimmer.Application.Example.tsx]: I don't even remember why I used `../../DetailsList`

**Comment by cliffkoh** [comment]
We should probably drop an email to the v-team informing of this breaking change "technicality". I am fine/supportive with making this kind of change in a minor bump otherwise (treating it as a bug fix).

File: packages/office-ui-fabric-react/src/Shimmered.ts [hunk 3]
```diff
@@ -0,0 +1 @@
+export * from './components/DetailsList/ShimmeredDetailsList';
```
Comment (cliffkoh) [inline packages/office-ui-fabric-react/src/Shimmered.ts]: Should the top level bundle be called ShimmeredDetailsList? (unless we anticipate adding more shimmered components here over time?)

Comment (dzearing) [reply packages/office-ui-fabric-react/src/Shimmered.ts]: I thought the same thing

Comment (dzearing) [reply packages/office-ui-fabric-react/src/Shimmered.ts]: ping @ThomasMichon

**Comment by dzearing** [comment]
Ping @ThomasMichon can you please edit your filename and resolve conflicts?

@Vitalius1 Once that is complete, can you help review and get this checked in?

Thank you!

**Pull request merged**